src/coin-selection/lib/Convex/MockChain/CoinSelection.hs

```diff
@@ -34,54 +34,56 @@ import           Data.Functor              (($>))
 {-| Balance and submit a transaction using the wallet's UTXOs
 on the mockchain, using the default network ID
 -}
-balanceAndSubmit :: (MonadMockchain m, MonadError BalanceTxError m) => Tracer m TxBalancingMessage -> Wallet -> TxBuilder -> m (Either SendTxFailed (C.Tx CoinSelection.ERA))
-balanceAndSubmit dbg wallet tx = do
+balanceAndSubmit :: (MonadMockchain m, MonadError BalanceTxError m, MonadFail m) => Tracer m TxBalancingMessage -> Wallet -> TxBuilder -> [C.ShelleyWitnessSigningKey] -> m (Either SendTxFailed (C.Tx CoinSelection.ERA))
```

Comment:
We are starting to have too many parameters for `balanceAndSubmit`. And `Wallet` is essentially just a signing key. I'll add a ticket to make this API nicer. We can keep this as it is for now.

Reply:
I agree.